core/chains/evm/logpoller/orm_test.go

```diff
@@ -655,3 +655,133 @@ func BenchmarkLogs(b *testing.B) {
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       2,
+			BlockHash:      common.HexToHash("0x2"),
+			BlockNumber:    2,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{common.HexToHash("0x0001").Bytes()},
+			EventSig:       eventSigB,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0002"),
+			Data:           []byte(""),
+		},
+	}))
+	require.NoError(t, orm.InsertBlock(common.HexToHash("0x2"), 2, time.Now()))
+
+	logs, err = orm.SelectLogsWithSigsExcluding(eventSigA, eventSigB, 0, sourceAddr, 0, 3, 0)
+
+	require.NoError(t, err)
+	require.Len(t, *logs, 0)
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       3,
+			BlockHash:      common.HexToHash("0x3"),
+			BlockNumber:    3,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{common.HexToHash("0x0002").Bytes()},
+			EventSig:       eventSigA,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0003"),
+			Data:           []byte("requestID-0002 data"),
+		},
+	}))
+	require.NoError(t, orm.InsertBlock(common.HexToHash("0x3"), 3, time.Now()))
+	logs, err = orm.SelectLogsWithSigsExcluding(eventSigA, eventSigB, 0, sourceAddr, 0, 3, 0)
+	require.NoError(t, err)
+	require.Len(t, *logs, 1)
+	require.Equal(t, (*logs)[0].Data, []byte("requestID-0002 data"))
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       4,
+			BlockHash:      common.HexToHash("0x3"),
+			BlockNumber:    3,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{common.HexToHash("0x0003").Bytes()},
+			EventSig:       eventSigA,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0004"),
+			Data:           []byte("requestID-0003 data"),
+		},
+	}))
+
+	logs, err = orm.SelectLogsWithSigsExcluding(eventSigA, eventSigB, 0, sourceAddr, 0, 3, 0)
+	require.NoError(t, err)
+	require.Len(t, *logs, 2)
+	require.Equal(t, (*logs)[0].Data, []byte("requestID-0002 data"))
+	require.Equal(t, (*logs)[1].Data, []byte("requestID-0003 data"))
+
+	require.NoError(t, orm.InsertLogs([]logpoller.Log{
+		{
+			EvmChainId:     (*utils.Big)(th.ChainID),
+			LogIndex:       5,
+			BlockHash:      common.HexToHash("0x4"),
+			BlockNumber:    4,
+			BlockTimestamp: time.Now(),
+			Topics:         [][]byte{common.HexToHash("0x0003").Bytes()},
+			EventSig:       eventSigB,
+			Address:        sourceAddr,
+			TxHash:         common.HexToHash("0x0005"),
+			Data:           []byte("requestID-0003 data"),
+		},
+	}))
+	require.NoError(t, orm.InsertBlock(common.HexToHash("0x4"), 4, time.Now()))
+
+	logs, err = orm.SelectLogsWithSigsExcluding(eventSigA, eventSigB, 0, sourceAddr, 0, 4, 0)
+	require.NoError(t, err)
+	require.Len(t, *logs, 1)
+	require.Equal(t, (*logs)[0].Data, []byte("requestID-0002 data"))
```

Comment:
whenever I see *log my instinct it to say we should check to make sure log isn't a nil pointer... in this case, it can't be but we could make that more explicit by returning a slice instead of pointer to slice (see above)

Reply:
Fixed in [`aabaab6` (#8870)](https://github.com/smartcontractkit/chainlink/pull/8870/commits/aabaab6db23df531bc85527260f0c59e54ad56eb)